Generate.ps1

```diff
@@ -148,7 +148,7 @@ $job = @(
     "$VANILLA_ARGUMENTS --input-file=vanilla-tests/swagger/xml-tag-with-attribute-and-value.json --namespace=fixtures.complexstreamstylexmlserialization",
     "$VANILLA_ARGUMENTS --input-file=vanilla-tests/swagger/security-info.json --namespace=fixtures.securityinfo --use-key-credential",
     "$VANILLA_ARGUMENTS --input-file=vanilla-tests/swagger/special-header.json --namespace=fixtures.specialheader",
-    "$VANILLA_ARGUMENTS --input-file=vanilla-tests/swagger/required-fields-as-ctor-args-transformation.json --namespace=fixtures.requiredfieldsascotrargstransformation --required-fields-as-ctor-args=true --output-model-immutable --null-byte-array-maps-to-empty-array",
+    "$VANILLA_ARGUMENTS --input-file=vanilla-tests/swagger/required-fields-as-ctor-args-transformation.json --namespace=fixtures.requiredfieldsascotrargstransformation --required-fields-as-ctor-args=true --output-model-immutable --null-byte-array-maps-to-empty-array --modelerfour.flatten-models=false --client-flattened-annotation-target=NONE ",
```

Comment:
Why we change the config for vanilla test?

Reply:
`modelerfour.flatten-models` defaults to `true`, and m4 will generate an anonymous model for the flattened property with `clientFlatten` false. 

The initialization only happens for `--modelerfour.flatten-models=false`. This test didn't have flatten case before, and can also be used to test `required flatten + required-fields-as-ctor-args` case.

Reply:
test here: https://github.com/Azure/autorest.java/pull/2891/files#diff-1e1169627c38fd2e98bf825ac7ba63135a7e8a262f2f281379b61a885d81221f

Reply:
Oh, It's meant for flatten issue found in mgmt. 

I didn't notice there is difference between flatten in mgmt and dpg... If the config isn't valid here, I can revert this test.

Reply:
Yeah, flatten in vanilla would be `modelerfour.flatten-models=true` + `client-flattened-annotation-target=TYPE`

Reply:
OK. Flatten in dpg seems disabled. Then this case seems not valid. I'll remove it.